tests/meta/observations/test_balsamic_observations_api.py

```diff
@@ -188,3 +345,114 @@ def test_load_cancer_observations(
+    [
+        (
+            SeqLibraryPrepCategory.WHOLE_EXOME_SEQUENCING,
+            BalsamicObservationPanel.EXOME,
+            LoqusdbInstance.SOMATIC_EXOME,
+        ),
+        (
+            SeqLibraryPrepCategory.TARGETED_GENOME_SEQUENCING,
+            BalsamicObservationPanel.LYMPHOID,
+            LoqusdbInstance.SOMATIC_LYMPHOID,
+        ),
+        (
+            SeqLibraryPrepCategory.TARGETED_GENOME_SEQUENCING,
+            BalsamicObservationPanel.MYELOID,
+            LoqusdbInstance.SOMATIC_MYELOID,
+        ),
+    ],
+    ids=["Exome", "Lymphoid", "Myeloid"],
+)
+def test_panel_upload(
+    prep_category: SeqLibraryPrepCategory,
+    panel: BalsamicObservationPanel,
+    loqusdb_instance: LoqusdbInstance,
+    mocker: MockerFixture,
+):
+
+    # GIVEN that a known panel is set in LIMS
+    lims_api: LimsAPI = create_autospec(LimsAPI)
+    lims_api.capture_kit = Mock(return_value="file.bed")
+    lims_api.get_source = Mock(return_value="Not valid")
+
+    # GIVEN a store
+    store_session = create_autospec(Session)
+    store: Store = create_autospec(Store, session=store_session)
+    bed = create_autospec(Bed)
+    bed.name = panel
+    store.get_bed_version_by_file_name_strict = Mock(
+        return_value=create_autospec(BedVersion, bed=bed, filename="file.bed")
+    )
+    store.get_bed_version_by_short_name = Mock(
+        return_value=create_autospec(BedVersion, bed=bed, filename="file.bed")
+    )
+
+    # GIVEN a panel case with a TGS sample
+    customer = create_autospec(Customer, internal_id=CustomerId.CUST110)
+    sample: Sample = create_autospec(Sample, prep_category=prep_category, loqusdb_id=None)
+    case: Case = create_autospec(
+        Case,
+        internal_id="case_id",
+        samples=[sample],
+        loqusdb_uploaded_samples=[],
+        customer=customer,
+    )
+    store.get_case_by_internal_id = Mock(return_value=case)
+    store.get_sample_ids_by_case_id = Mock(return_value=[sample])
+
+    # GIVEN balsamic observations API
+    balsamic_observations_api = BalsamicObservationsAPI(
+        create_autospec(
+            CGConfig,
+            lims_api=lims_api,
+            balsamic=Mock(),
+            loqusdb=Mock(),
+            loqusdb_rd_lwp=Mock(),
+            loqusdb_wes=Mock(),
+            loqusdb_somatic=Mock(),
+            loqusdb_tumor=Mock(),
+            loqusdb_somatic_lymphoid=Mock(),
+            loqusdb_somatic_myeloid=Mock(),
+            loqusdb_somatic_exome=Mock(),
+            run_instruments=create_autospec(
+                RunInstruments,
+                illumina=create_autospec(IlluminaConfig, demultiplexed_runs_dir="runs_dir"),
+            ),
+            sentieon_licence_server=Mock(),
+            status_db=store,
+        ),
+    )
+
+    loqusdb_selection = mocker.spy(ObservationsAPI, "get_loqusdb_api")
+    loqusdb_load = mocker.patch.object(LoqusdbAPI, "load")
+    mocker.patch.object(LoqusdbAPI, "get_case", side_effect=[None, {LOQUSDB_ID: 1}])
```

Comment:
Tricky to see what is going on here. Are there any alternative solutions? The issue is that the first call needs to be None.